src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -149,28 +149,18 @@ public AvlNodeHead(int hashCode, K key, V value, Node next)
             public override Node Next => next;
         }
 
-        // separate class to ensure that HashCode field 
-        // is layed out in ram  before other AvlNode fields
-        private abstract class HashedNode : Node
```

Comment:
I think I will move balance to the base, to keep both benefits of ordering and packing, however small they could be.